pest/src/position.rs

```diff
@@ -410,7 +409,7 @@ impl<'i> Ord for Position<'i> {
 
 impl<'i> Hash for Position<'i> {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        (self.input as *const [u8]).hash(state);
+        (self.input.as_bytes() as *const [u8]).hash(state);
```

Comment:
I believe this approach only hashes the first hash of the `&str`/fat pointer.

```suggestion
        (self.input as *const str).hash(state);
```